packages/opentelemetry-core/src/context/propagation/composite.ts

```diff
@@ -81,4 +81,14 @@ export class CompositePropagator implements TextMapPropagator {
       return ctx;
     }, context);
   }
+
+  fields() {
+    const fields = new Set<string>();
```

Comment:
why do you create a set and then again convert this into array. Can you simply define fields as an array and then do 
```
  for (const child of this._propagators) {
      for (const field of child.fields()) {
        fields.push(field);
      }
    }
```

Reply:
To remove duplicates

Reply:
Moved logic to the constructor to remove duplicates without incurring the cost of the new Set on each call to fields. https://github.com/open-telemetry/opentelemetry-js/pull/1615/files#diff-7c180f9fb8af83fee181f9353da5df99459bdf2574127df24b97e3060ba4e4e0R41